### ISSUE TITLE
Gopass does not list empty folders and rm -r allows to create them

### ISSUE THREAD

**Issue by AnomalRoil** [gopasspw/gopass#1723]
### Summary

If one creates an entry in a subfolder of a subfolder and deletes the last subfolder using `rm -r`, even if the first subfolder is empty, it won't be deleted, whereas `gopass rm entry` does clean the folders in the entry path if they are empty.

## To reproduce

```
gopass generate new/folder/test/hello 10
gopass ls new
gopass rm -r new/folder/test
gopass ls new
> Error: Entry 'new' not found
ls $(gopass config path | cut -d ' ' -f 2)/new
> folder
gopass generate new/folder/test/hello 10
gopass rm new/folder/test/hello 
ls $(gopass config path | cut -d ' ' -f 2)/new
> ls: cannot access '/home/anomalroil/.local/share/gopass/stores/root/new': No such file or directory
```

**Comment by dominikschulz** [comment]
Interesting. What course of action would you propose?

* `gopass rm` should likely remove empty folders when removing nested entries?! Shouldn't be too hard to do.
* `gopass fsck` could probably look for these as well, since people might have them lying around already. Should be easy.
* Should `gopass ls` display empty folders? This might be tricky.

**Comment by AnomalRoil** [comment]
I agree with you for the 3 items. Let's have `rm` delete them when it sees the parent directory is empty and `fsck` detect empty folders :)

**Comment by dominikschulz** [comment]
I don't plan to work on `gopass ls`, but the other items should be fixed in #1725.
